hscontrol/app.go

```diff
@@ -931,6 +922,58 @@ func notFoundHandler(
 	writer.WriteHeader(http.StatusNotFound)
 }
 
+func (h *Headscale) loadACLPolicy() error {
+	switch h.cfg.ACL.PolicyMode {
+	case types.ACLPolicyModeFile:
+		if h.cfg.ACL.PolicyPath != "" {
+			aclPath := util.AbsolutePathFromConfigPath(h.cfg.ACL.PolicyPath)
+			pol, err := policy.LoadACLPolicyFromPath(aclPath)
+			if err != nil {
+				return pkgerrors.Wrap(err, "failed to load ACL policy from file")
+			}
+
+			h.ACLPolicy = pol
+
+			log.Info().
+				Str("path", aclPath).
+				Msg("ACL policy successfully loaded, notifying nodes of change")
+		}
+	case types.ACLPolicyModeDB:
+		acl, err := h.db.GetACL()
+		if err != nil {
+			if db.IsNotFoundError(err) {
+				log.Info().Msg("No ACL policy found in database")
+				return nil
+			}
+
+			return pkgerrors.Wrap(err, "failed to fetch ACL policy from database")
+		}
+
+		pol, err := policy.LoadACLPolicyFromBytes(acl.Policy, "json")
+		if err != nil {
+			return pkgerrors.Wrap(err, "failed to parse policy")
+		}
+
+		h.ACLPolicy = pol
+
+		log.Info().
+			Msg("ACL policy successfully reloaded, notifying nodes of change")
+	default:
+		log.Warn().
+			Str("mode", string(h.cfg.ACL.PolicyMode)).
+			Msg("Unknown ACL policy mode")
+	}
+
+	if h.ACLPolicy != nil {
+		ctx := types.NotifyCtx(context.Background(), "acl-sighup", "na")
```

Comment:
Do we want to notifyCtx + notifyAll if it's not a reload, but a normal startup? 🤔

Reply:
We do not. 😅 Thanks, I've updated this bit.